src/leappto/cli.py

```diff
@@ -135,27 +143,77 @@ def _set_ssh_config(username, identity, use_sshpass=False):
         return use_sshpass, ssh_options
 
     class MigrationContext:
-        def __init__(self, target, ssh_settings, disk, forwarded_ports=None):
+
+        SOURCE = 'source'
+        TARGET = 'target'
+
+        _SSH_CTL_PATH = '{}/.ssh/ctl'.format(os.environ['HOME'])
+        _SSH_CONTROL_PATH = '-o ControlPath="{}/%L-%r@%h:%p"'.format(_SSH_CTL_PATH)
+
+        def __init__(self, target, target_ssh_cfg, disk, source=None, source_ssh_cfg=None, forwarded_ports=None,
+                    rsync_cp_backend=False):
+            self.source = source
             self.target = target
-            self.use_sshpass, self.target_cfg = ssh_settings
+            self.source_use_sshpass, self.source_cfg = (None, None) if source_ssh_cfg is None else source_ssh_cfg
+            self.target_use_sshpass, self.target_cfg = target_ssh_cfg
             self._cached_ssh_password = None
             self.disk = disk
+            self.rsync_cp_backend = rsync_cp_backend
             if forwarded_ports is None:
                 forwarded_ports = [(80, 80)]  # Default to forwarding plain HTTP
             else:
                 forwarded_ports = list(forwarded_ports)
             self.forwarded_ports = forwarded_ports
 
+        def __get_machine_opt_by_context(self, machine_context):
+            return (getattr(self, '{}_{}'.format(machine_context, opt)) for opt in ['addr', 'cfg', 'use_sshpass'])
```

Comment:
I don't like that, but I didn't want to change a lot of functions and it was better to use class attr, we need to refactor that class.